src/Servers/Kestrel/Core/src/Internal/Http/ChunkWriter.cs

```diff
@@ -16,16 +16,8 @@ internal static class ChunkWriter
         public static int BeginChunkBytes(int dataCount, Span<byte> span)
         {
             // Determine the most-significant non-zero nibble
-            int total, shift;
-            var count = dataCount;
-            total = (count > 0xffff) ? 0x10 : 0x00;
-            count >>= total;
-            shift = (count > 0x00ff) ? 0x08 : 0x00;
-            count >>= shift;
-            total |= shift;
-            total |= (count > 0x000f) ? 0x04 : 0x00;
-
-            count = (total >> 2) + 3;
+            int total, shift; 
+            var count = GetBeginChunkByteCount(dataCount, out total, out shift);
```

Comment:
It doesn't look like the shift out param is actually being used given that it get's reassigned at the beginning of the for loop:

```C#
for (shift = total; shift >= 0; shift -= 4)
```

Reply:
The only reason I kept it was to reuse the int.

Reply:
I wouldn't worry about it unless there's some evidence that we have so many locals that they're spilling out of registers.